recipes/libsystemd/all/conanfile.py

```diff
@@ -152,4 +152,6 @@ def package(self):
 
     def package_info(self):
         self.cpp_info.libs = ["systemd"]
+        # systemd uses only major version in its .pc file
+        self.cpp_info.version = self.version.split('.')[0]
```

Comment:
I'm afraid this approach is not fine as it mixes global properties (`options.version`) with generator specific properties .
Maybe conan needs `self.cpp_info.version["pkg_config"]`?

[Please open an issue at the conan repo if this feature is missing.](https://github.com/conan-io/conan/issues)
(I haven't tested this yet)

Reply:
I'm not sure that it is generator specific problem. I mentioned pkg-config only because it is the only generator supported by systemd.
The original problem is that the source package version `246.6` is not matches with binary package version `246` (and not matches with library soname version `0.29.0`).

Maybe it would be better to change source package version from `246.6` to `246`. Something like this:
```
sources:
  "246":
    url: "https://github.com/systemd/systemd-stable/archive/v246.6.tar.gz"
```
instead of
```
sources:
  "246.6":
    url: "https://github.com/systemd/systemd-stable/archive/v246.6.tar.gz"
```